### ISSUE TITLE
Cannot get source files from BC15

### ISSUE THREAD

**Issue by larswestman** [microsoft/navcontainerhelper#577]
The folder 15.0.35659.0-w1-al is empty after container creation with -includeAL and I get this error:

Copying Al Source Files from C:\ProgramData\NavContainerHelper\Extensions\Original-15.0.35659.0-w1-al\* to C:\Repos\BC Master\al
Get-Content : Cannot find path 'C:\ProgramData\NavContainerHelper\Extensions\Original-15.0.35659.0-w1-al\app.json' because it does not exist.

I've uninstalled all versions of NAVcontainerhelper and installed it again. Then I restarted Powershell ISE.

**This is the script:**

$imageName = "bcinsider.azurecr.io/bcsandbox-master:latest-ltsc2019"
$containerName = "bc-master"
$auth = "UserPassword"
$credential = New-Object pscredential 'admin', (ConvertTo-SecureString -String 'P@ssword1' -AsPlainText -Force)
$licenseFile = "C:\licenses\dev license.flf"

docker login "bcinsider.azurecr.io" -u "Username" -p "Password"

New-BCContainer -accept_eula `
                -imageName $imageName `
                -containerName $containerName `
                -auth $auth `
                -credential $credential `
                -licenseFile $licenseFile `
                -updateHosts `
                -includeAL `
                -useBestContainerOS

$alProjectFolder = "C:\Repos\BC Master\al"
$myModifiedObjects = @()

$alFileStructure = { Param ([string] $type, [int] $id, [string] $name) 
    if ($myModifiedObjects.Contains("$type$id")) {
        $folder = "Modified"
    }
    elseif (($id -ge 50000 -and $id -le 59999) -or ($id -gt 1999999999)) {
        $folder = "My"
    }
    else {
        $folder = "BaseApp"
    }
    switch ($type) {
            "enum"           { "$folder\Enums\$($name).al" }
            "enumextension"  { "$folder\Enums\$($name).al" }
            "page"           { "$folder\Pages\$($name).al" }
            "pageextension"  { "$folder\Pages\$($name).al" }
            "table"          { "$folder\Tables\$($name).al" }
            "tableexension"  { "$folder\Tables\$($name).al" }
            "codeunit"       { "$folder\Codeunits\$($name).al" }
            "query"          { "$folder\Queries\$($name).al" }
            "report"         { "$folder\Reports\$($name).al" }
            "xmlport"        { "$folder\XmlPorts\$($name).al" }
            "profile"        { "$folder\Profiles\$($name).al" }
            "dotnet"         { "$folder\$($name).al" }
            ".rdlc"          { "$folder\Layouts\$($name).rdlc" }
            ".docx"          { "$folder\Layouts\$($name).docx" }
            ".xlf"           { "$folder\$($name).xlf" }
            default { throw "Unknown type '$type'" }
        }
}
Create-AlProjectFolderFromNavContainer -containerName $containerName `
                                       -alProjectFolder $alProjectFolder `
                                       -useBaseLine `
                                       -name "Base Application" `
                                       -publisher "Microsoft" `
                                       -version "15.0.0.0" `
                                       -alFileStructure $alFileStructure

**And here's the output:**

Login Succeeded
NavContainerHelper is version 0.6.3.1
Host is Microsoft Windows 10 Pro - 1903
Docker Client Version is 19.03.1
Docker Server Version is 19.03.1
Using image bcinsider.azurecr.io/bcsandbox-master:latest-ltsc2019
Removing container bc-master
Removing bc-master from hosts
Removing C:\ProgramData\NavContainerHelper\Extensions\bc-master
Creating Container bc-master
Using license file C:\licenses\dev license.flf
Version: 15.0.35659.0-w1
Platform: 15.0.35510.0
Generic Tag: 0.0.9.92
Container OS Version: 10.0.17763.678 (ltsc2019)
Host OS Version: 10.0.18362.295 (1903)
A better Generic Container OS exists for your host (1903)
Using generic image mcr.microsoft.com/dynamicsnav:generic-1903
Generic Container OS Version: 10.0.18362.295 (1903)
Generic Tag of better generic: 0.0.9.92
Using locale en-US
Using process isolation
Disabling the standard eventlog dump to container log every 2 seconds (use -dumpEventLog to enable)
Files in C:\ProgramData\NavContainerHelper\Extensions\bc-master\my:
- AdditionalOutput.ps1
- license.flf
- MainLoop.ps1
- updatehosts.ps1

Creating container bc-master from image mcr.microsoft.com/dynamicsnav:generic-1903
55846d0b04407d64dd8969a4625b9141b0a8776e714dbdbdc8082b743ad83f68
Waiting for container bc-master to be ready
Installing Business Central
Installing Url Rewrite
Installing OpenXML
Installing DotNetCore
Starting Local SQL Server
Starting Internet Information Server
Copying Service Tier Files
Copying Web Client Files
Copying Client Files
Copying ModernDev Files
Copying PowerShell Scripts
Copying ConfigurationPackages
Copying Test Assemblies
Copying ReportBuilder
Changing Database Server Collation to Latin1_General_100_CS_AS
SQL Server 2017 transmits information about your installation experience, as well as other usage and performance data, to Microsoft to help improve the product. To learn more about 
SQL Server 2017 data processing and privacy controls, please see the Privacy Statement.
Copying Cronus database
Modifying Business Central Service Tier Config File for Docker
Creating Business Central Service Tier
Installing SIP crypto provider: 'C:\Windows\System32\NavSip.dll'
Starting Business Central Service Tier
Installation took 243 seconds
Installation complete
Initializing...
Starting Container
Hostname is bc-master
PublicDnsName is bc-master
Using NavUserPassword Authentication
Creating Self Signed Certificate
Self Signed Certificate Thumbprint EA049C79092727D421B2FD4D25CB84950949AE18
Modifying Service Tier Config File with Instance Specific Settings
Restarting Service Tier
Registering event sources
Creating DotNetCore Web Server Instance
Enabling Financials User Experience
Using license file 'c:\run\my\license.flf'
Import License
Creating http download site
Setting SA Password and enabling SA
Creating lars as SQL User and add to sysadmin
WARNING: This license is not compatible with this version of Business Central.
Creating SUPER user
WARNING: This license is not compatible with this version of Business Central.
WARNING: This license is not compatible with this version of Business Central.
Container IP Address: 172.25.36.198
Container Hostname  : bc-master
Container Dns Name  : bc-master
Web Client          : http://bc-master/BC/
Dev. Server         : http://bc-master
Dev. ServerInstance : BC

Files:
http://bc-master:8080/al-4.0.166043.vsix

Initialization took 75 seconds
Ready for connections!
Reading CustomSettings.config from bc-master
Creating Desktop Shortcuts for bc-master
Creating .net Assembly Reference Folder for VS Code
Copying DLLs from C:\Windows\assembly to assemblyProbingPath
Copying DLLs from C:\Program Files\Microsoft Dynamics NAV\150\Service to assemblyProbingPath
Copying DLLs from C:\Program Files (x86)\Microsoft Dynamics NAV\150\RoleTailored Client to assemblyProbingPath
Copying DLLs from C:\Program Files (x86)\Open XML SDK to assemblyProbingPath
Container bc-master successfully created
Copying Al Source Files from C:\ProgramData\NavContainerHelper\Extensions\Original-15.0.35659.0-w1-al\* to C:\Repos\BC Master\al
Get-Content : Cannot find path 'C:\ProgramData\NavContainerHelper\Extensions\Original-15.0.35659.0-w1-al\app.json' because it does not exist.
At C:\Program Files\WindowsPowerShell\Modules\navcontainerhelper\0.6.3.1\AppHandling\Create-AlProjectFolderFromNavContainer.ps1:122 char:20
+         $appJson = Get-Content "$alFolder\app.json" | ConvertFrom-Jso ...
+                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : ObjectNotFound: (C:\ProgramData\...-w1-al\app.json:String) [Get-Content], ItemNotFoundException
+ FullyQualifiedErrorId : PathNotFound,Microsoft.PowerShell.Commands.GetContentCommand

**Comment by freddydk** [comment]
It is likely because the folder was created during an unsuccessful attempt to create the container - then it just looks whether the folder is there.
Maybe I should check whether it is empty.
If you delete the empty folder and retry, then it should work.

**Comment by larswestman** [comment]
It seems as we get one step further but then I get this error:

Cannot find path 'C:\Applications\' because it does not exist.
at <ScriptBlock>, <No file>: line 7
Cannot find path 'C:\Applications\' because it does not exist.
At C:\Program Files\WindowsPowerShell\Modules\navcontainerhelper\0.6.3.1\ContainerHandling\Invoke-ScriptInNavContainer.ps1:37 char:13
+             Invoke-Command -Session $session -ScriptBlock $scriptbloc ...
+             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : ObjectNotFound: (C:\Applications\:String) [Get-ChildItem], ItemNotFoundException
+ FullyQualifiedErrorId : PathNotFound,Microsoft.PowerShell.Commands.GetChildItemCommand
+ PSComputerName        : 88f6d726f3f1f88a45eec1f6f16d2b82e2c155941584a1fd0d90e91f262c9ca0

**Comment by freddydk** [comment]
Isn't there an Applications folder in this folder: C:\ProgramData\NavContainerHelper\15.0.35659.0-w1-Files
Else you might have tried to run this with an old navcontainerhelper and extracted the content without the applications folder.
Then you need to remove that folder as well.

**Comment by larswestman** [comment]
On my first re-run after deleting the folder 15.0.35659.0-w1-Files I could see that the sub folder Applications was created and I got the app-file created. I still got an error about app.json missing though since Original-15.0.35659.0-w1-al was empty.

On my second re-run I deleted both folders and it worked.

I guess one error was that I had been running on an old version of NAVcontainerhelper and 15.0.35659.0-w1-Files wasn't in good shape. Next error seems to be that Original-15.0.35659.0-w1-al must be deleted

At last/least I got my al files :-) But it was a bit messy...
